plugins/org.jkiss.dbeaver.ui/src/org/jkiss/dbeaver/ui/controls/CustomFormEditor.java

```diff
@@ -301,16 +305,47 @@ public void widgetSelected(SelectionEvent e) {
             link.setLayoutData(new GridData(GridData.FILL_HORIZONTAL));
             return link;
         } else if (isTextPropertyType(propType)) {
-            if (property instanceof ObjectPropertyDescriptor && ((ObjectPropertyDescriptor) property).getLength() == PropertyLength.MULTILINE) {
+            if (property instanceof ObjectPropertyDescriptor && property.getLength() == PropertyLength.MULTILINE) {
                 Label label = UIUtils.createControlLabel(parent, propertyDisplayName);
                 label.setLayoutData(new GridData(GridData.VERTICAL_ALIGN_BEGINNING));
-                Text editor = new Text(parent, SWT.BORDER | SWT.MULTI | SWT.V_SCROLL | SWT.WRAP | (readOnly ? SWT.READ_ONLY : SWT.NONE));
-
+                Composite composite = UIUtils.createComposite(parent, 2);
+                Text editor = new Text(composite, SWT.MULTI | SWT.WRAP | SWT.BORDER | (readOnly ? SWT.READ_ONLY : SWT.NONE));
+                editor.setLayoutData(new GridData(GridData.FILL_BOTH));
                 editor.setText(objectValueToString(value));
-                GridData gd = new GridData(GridData.FILL_HORIZONTAL);
-                // Make multiline editor at least two lines height
-                gd.heightHint = (UIUtils.getTextHeight(editor) + editor.getBorderWidth()) * 2;
-                editor.setLayoutData(gd);
+
+                int textHeight = UIUtils.getTextHeight(editor) * 3;
+                GridData gdEditor = new GridData(SWT.FILL, SWT.TOP, true, false);
+                gdEditor.heightHint = textHeight;
+                composite.setLayoutData(gdEditor);
+
+                Button editButton = new Button(composite, SWT.FLAT | SWT.PUSH);
+                editButton.setImage(DBeaverIcons.getImage(UIIcon.DOTS_BUTTON));
```

Comment:
While we are at it, please add a tooltip for this button so its intention is clear without requiring a click.

Reply:
fix